jax/config.py

```diff
@@ -138,6 +148,17 @@ def disable_omnistaging(self):
         disabler()
       self.omnistaging_enabled = False
 
+  @property
+  def x64_enabled(self):
+    if _thread_local_state.enable_x64 is not None:
+      return bool(_thread_local_state.enable_x64)
+    else:
+      return bool(self.read('jax_enable_x64'))
```

Comment:
Should we set the thread-local state on first read if it doesn't exist?

Reply:
My reason for not setting it is to avoid breaking current behavior in cases where the user is setting the flag at runtime. But I suppose that pattern is broken to begin with...

I'll make the change.